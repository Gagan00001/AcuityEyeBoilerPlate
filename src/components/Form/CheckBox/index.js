import React from 'react';
import {
  Form, Checkbox, Row, Col,
} from 'antd';
import PropTypes from 'prop-types';
import './checkbox.scss';
import classNames from 'classnames';

const CheckBox = ({
  name,
  className,
  label,
  labelSpan,
  inputSpan,
  required,
  checked,
  id,
  onClick,
  key,
  component: Component,
  options,
  labelAfter,
  children,
  onChange,
  disabled,
  onBlur,
  dataTestId,
  checkBoxProps,
  ...otherProps
}) => (
  <div className="custom-checkbox">
    <Row>
      <Col span={labelSpan} order={labelAfter && 1} className={classNames({ space: labelAfter })}>
        <div className="ant-form-item-label ant-form-item-label-left">
          <label
            className={classNames('ant-form-item-no-colon')}
            title={label}
            htmlFor={id}
          >
            {label}
          </label>
          <span className={classNames('req-star', required && 'ant-form-item-required')} />
        </div>
      </Col>
      <Col span={inputSpan}>
        <Form.Item
          className={className}
          name={name}
          required
          {...otherProps}
        >
          <Component
            onClick={onClick}
            onChange={onChange}
            checked={checked}
            id={id}
            key={key}
            options={options}
            disabled={disabled}
            onBlur={onBlur}
            data-testid={dataTestId || name}
            {...checkBoxProps}
          >
            {children}
          </Component>
        </Form.Item>
      </Col>
    </Row>
  </div>
);

CheckBox.Group = Checkbox.Group;

CheckBox.defaultProps = {
  name: '',
  label: '',
  onClick: (event) => event.stopPropagation(),
  required: false,
  checked: false,
  labelSpan: 10,
  inputSpan: 14,
  component: Checkbox,
  className: '',
  id: '',
  options: [],
  key: '',
  labelAfter: false,
  children: [],
  onChange: () => { /* This is intentional */ },
  disabled: false,
  onBlur: () => { /* This is intentional */ },
  dataTestId: '',
  checkBoxProps: {},
};

CheckBox.propTypes = {
  name: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
  ]),
  label: PropTypes.string,
  onClick: PropTypes.func,
  required: PropTypes.bool,
  checked: PropTypes.bool,
  labelSpan: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  inputSpan: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  component: PropTypes.element,
  className: PropTypes.string,
  id: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  options: PropTypes.instanceOf(Array),
  key: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  labelAfter: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
  onBlur: PropTypes.func,
  dataTestId: PropTypes.string,
  checkBoxProps: PropTypes.instanceOf(PropTypes.object),
};

export default CheckBox;
