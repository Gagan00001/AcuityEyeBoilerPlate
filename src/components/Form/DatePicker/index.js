import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import {
  Form, DatePicker as DatePickerComponent, Row, Col,
} from 'antd';
import cloneDeep from 'lodash/cloneDeep';

import './datePicker.scss';

const DatePicker = (props) => {
  const {
    label,
    name,
    placeholder,
    showTime,
    dateFormat,
    tabIndex,
    required,
    rules,
    onChange,
    value,
    className,
    inputSpan,
    labelSpan,
    disabled,
    id,
    disabledDate,
    picker,
    shouldShowWarningText,
    allowClear,
    Component,
    datePickerProps,
    dataTestId,
    ...otherProps
  } = props;

  const clonedRules = cloneDeep(rules);
  if (required) {
    clonedRules.push({
      required: true,
      message: shouldShowWarningText ? `${label} is required` : ' ',
    });
  }
  return (
    <div className="custom-picker">
      <Row>
        <Col span={labelSpan}>
          <div className="ant-form-item-label ant-form-item-label-left">
            <label
              className={classNames('ant-form-item-no-colon')}
              title={label}
              htmlFor={id}
            >
              {label}
            </label>
            <span className={classNames('req-star', required && 'ant-form-item-required')} />
          </div>
        </Col>
        <Col span={inputSpan}>
          <Form.Item
            name={name}
            rules={[...clonedRules]}
            {...otherProps}
          >
            <Component
              value={value}
              defaultValue={value}
              picker={picker}
              placeholder={placeholder}
              format={dateFormat}
              showTime={showTime}
              tabIndex={tabIndex}
              onChange={onChange}
              className={classNames('date-picker', className)}
              disabled={disabled}
              disabledDate={disabledDate}
              allowClear={allowClear}
              data-testid={dataTestId}
              {...datePickerProps}
            />
          </Form.Item>
        </Col>
      </Row>
    </div>
  );
};

DatePicker.defaultProps = {
  labelSpan: 10,
  inputSpan: 14,
  name: '',
  label: '',
  required: false,
  showTime: false,
  dateFormat: ['MM/DD/YYYY', 'MMDDYYYY', 'M/D/YYYY'],
  onChange: () => { /* This is intentional */ },
  disabled: false,
  disabledDate: () => { /* This is intentional */ },
  rules: [],
  className: '',
  id: '',
  tabIndex: 0,
  value: '',
  picker: 'date',
  shouldShowWarningText: true,
  Component: DatePickerComponent,
  allowClear: true,
  datePickerProps: {},
};

DatePicker.propTypes = {
  labelSpan: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  inputSpan: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  onChange: PropTypes.func,
  name: PropTypes.string,
  className: PropTypes.string,
  label: PropTypes.string,
  required: PropTypes.bool,
  allowClear: PropTypes.bool,
  id: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  showTime: PropTypes.bool,
  dateFormat: PropTypes.instanceOf(Array),
  disabled: PropTypes.bool,
  rules: PropTypes.instanceOf(Array),
  disabledDate: PropTypes.func,
  tabIndex: PropTypes.number,
  value: PropTypes.string,
  picker: PropTypes.string,
  shouldShowWarningText: PropTypes.bool,
  Component: PropTypes.node,
  datePickerProps: PropTypes.instanceOf(Object),
};

export default DatePicker;
