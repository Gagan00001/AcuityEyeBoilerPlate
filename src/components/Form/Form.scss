@import "../../css/vars.scss";

@mixin basic-input-ui {
  height: 2.8rem;
  border-radius: $gap-04;
  color: $color-blue-light;
  font-size: $font-size-m;
}

.form-grid-component {
  display: grid;
  grid-row-gap: 10px;

  .form-grid-column {
    display: grid;
    grid-auto-rows: auto;  // 28px
    grid-template-columns: 1fr;
    grid-row-gap: 0px; //10px
  }

  .form-grid-row {
    display: grid;
    grid-auto-columns: 28px;
    grid-template-rows: 1fr;
    grid-column-gap: 10px;
  }

  
  .ant-form-item-label {
    white-space: normal;
    overflow: visible;
    label {
      display: flex;
      color: $color-blue-light;
      font-size: $font-size-xsm;
    }
    .ant-form-item-required::after {
      content: "*";
      color: $color-red;
    }
  }
    .text-input {
      @include basic-input-ui;
    }
    .select-box {
      .ant-select-selector {
        @include basic-input-ui;
        .ant-select-selection-search {
          input {
            height: 2.6rem;
            font-size: $font-size-m;
          }
        }
      }
    }
    .date-picker {
      @include basic-input-ui;
    }


  .form-item-rowspan > .ant-form-item-label > label {
    align-items: flex-start;
    padding-top: $gap-02;
    line-height: 1.4;
  }
  .ant-form-item-has-error {
    .ant-form-item-control-input-content {
      .ant-input {
        border-color: $color-border-red;
      }
      .ant-picker {
        border-color: $color-border-red;
      }
    }
  }
}
.fields-wrapper {
  .ant-form-item-label {
    white-space: normal;
    overflow: visible;
    label {
      display: flex;
      color: $color-blue-light;
      font-size: $font-size-m;
    }
    .ant-form-item-required::after {
      content: "*";
      color: $color-red;
    }
  }

}
.cursor-pointer{
  cursor: pointer;
}
