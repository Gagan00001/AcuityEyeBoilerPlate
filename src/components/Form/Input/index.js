import React, { useCallback } from "react";
import { Form, Input, Row, Col } from "antd";
import PropTypes from "prop-types";
import "./index.scss";
import cloneDeep from "lodash/cloneDeep";
import classNames from "classnames";

const validationChecker = (condition, errorMessage) => {
  if (condition) {
    return Promise.resolve();
  }
  return Promise.reject(errorMessage);
};

const alphanumericRegex = (decimalPlaces = 0, addNegative = "") => {
  const decimalRegex = `(\\.[0-9]{0,${decimalPlaces}})?`;
  return new RegExp(
    `^${addNegative}[a-zA-Z0-9${decimalPlaces ? "" : " "}]*${
      decimalPlaces ? decimalRegex : ""
    }$`
  );
};
const numbersWithSpecialCharsRegex = /^[0-9#?!@$%^&*-/]+$/;
const numberOnlySignRegex = (decimalPlaces) =>
  new RegExp(`^-?[0-9]*(.[0-9]{1,${decimalPlaces || 2}})?$`);
const numberWithHyphenRegex = /^[0-9-]+$/;
const numberWithDecimalPlaces = /^\d{0,8}(\.\d{1,2})?$/;
const valueAfterDecimal = /^\d*\.?\d+$/;
const numberRegex = (decimalPlaces = 0, addNegative = "") => {
  const decimalRegex = `(\\.[0-9]{0,${decimalPlaces}})?`;
  return new RegExp(
    `^${addNegative}[0-9]+${decimalPlaces ? decimalRegex : ""}$`
  );
};
const naturalNumberRegex = /^[1-9]\d*$/;
const passwordRegex = /(?=.*\d)/;
const updatePasswordRegex =
  /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?\d)(?=.*?[#?!@$%^&*-]).{8,15}$/;
const passwordValidator = (rules, value) =>
  !value
    ? Promise.resolve()
    : validationChecker(
        passwordRegex.test(value.trim()),
        "The password must contain at least 1 numeric character"
      );
const updatePasswordValidator = (rules, value) =>
  !value
    ? Promise.resolve()
    : validationChecker(
        updatePasswordRegex.test(value.trim()),
        "The password must contain at least 1 numeric, 1 lowercase, 1 uppercase and 1 special character and should be of minimum 8 characters"
      );

const numWithSpecialCharValidator = (rules, value) =>
  !value
    ? Promise.resolve()
    : validationChecker(
        numbersWithSpecialCharsRegex.test(value.trim()),
        "Value should contain numbers and special characters only"
      );
const stringValidator = (rules, value) =>
  !value
    ? Promise.resolve()
    : validationChecker(true, "Value should contain characters only");
const alphaNumericValidator =
  (decimalPlaces, isNegative = false) =>
  (rules, value) => {
    if (!value) {
      return Promise.resolve();
    }
    if (isNegative) {
      return validationChecker(
        alphanumericRegex(decimalPlaces, "-?").test(value),
        <ul className="error-type-list">
          <li>Positive/negative values.</li>
          <li>
            <div>Number of decimals: {decimalPlaces}.</div>
          </li>
          <li>Space not allowed.</li>
        </ul>
      );
    }
    return validationChecker(
      alphanumericRegex().test(value.trim()),
      "Value should be alphanumeric"
    );
  };
const numberWithDecimalPlacesValidator = (rules, value) =>
  !value
    ? Promise.resolve()
    : validationChecker(
        numberWithDecimalPlaces.test(value.trim()),
        "Amount format should be XXXXXXXX.XX"
      );
const numberValidator =
  (decimalPlaces, isNegative = false) =>
  (rules, value) => {
    if (!value) {
      return Promise.resolve();
    }
    let errorMsg = "Value should contain numbers only";
    if (decimalPlaces) {
      errorMsg = `Value should contain numbers only with upto ${decimalPlaces} decimal places`;
    }
    if (isNegative) {
      return validationChecker(
        numberRegex(decimalPlaces, "-?").test(value),
        errorMsg
      );
    }
    return validationChecker(numberRegex(decimalPlaces).test(value), errorMsg);
  };
const isValueAfterDecimalValidator = (rules, value) =>
  !value
    ? Promise.resolve()
    : validationChecker(
        valueAfterDecimal.test(value.trim()),
        "There should be numbers after decimal"
      );

const numberOnlySignValidator =
  (decimalPlaces = 0) =>
  (rules, value) =>
    !value
      ? Promise.resolve()
      : validationChecker(
          numberOnlySignRegex(decimalPlaces).test(value.trim()),
          "Value should be a valid number"
        );
const numberWithHyphenValidator = (rules, value) =>
  !value
    ? Promise.resolve()
    : validationChecker(
        numberWithHyphenRegex.test(value.trim()),
        "Value should be a valid"
      );
const naturalNumberValidator = (rules, value) =>
  !value
    ? Promise.resolve()
    : validationChecker(
        naturalNumberRegex.test(value.trim()),
        "Value should be a natural number"
      );
const requiredValidator = (label) => (rules, value) =>
  value && `${value}`.trim().length
    ? Promise.resolve()
    : validationChecker(
        value && `${value?.trim()}`?.length,
        `${label} is required`
      );
const handleKeyDown = (e) => {
  if (
    !(
      (e.key >= 0 && e.key <= 9) ||
      e.key === "." ||
      e.key === "Backspace" ||
      e.key === "ArrowLeft" ||
      e.key === "ArrowRight" ||
      e.key === "Delete" ||
      e.key === "Enter" ||
      e.key === "Tab" ||
      (e.ctrlKey && (e.key === "a" || e.key === "c"))
    )
  ) {
    e.preventDefault();
  }
};

const applyAddOnRules = (clonedRules, props) => {
  const {
    charOnly,
    alphaNumericOnly,
    numberOnly,
    numbersAndSpecialChars,
    decimalPlaces,
    type,
    applyValidation,
    numberOnlyWithSign,
    updatePassword,
    numberWithHyphen,
    numberWithDecimal,
    naturalNumberOnly,
    isValueAfterDecimal,
    required,
    label,
    allowNegative,
  } = props;
  if (isValueAfterDecimal) {
    clonedRules.push({ validator: isValueAfterDecimalValidator });
  }
  if (updatePassword) {
    clonedRules.push({ validator: updatePasswordValidator });
  }
  if (numberOnly) {
    clonedRules.push({
      validator: numberValidator(decimalPlaces, allowNegative),
    });
  }
  if (alphaNumericOnly) {
    clonedRules.push({
      validator: alphaNumericValidator(decimalPlaces, allowNegative),
    });
  }
  if (numberOnlyWithSign) {
    clonedRules.push({ validator: numberOnlySignValidator(decimalPlaces) });
  }
  if (numbersAndSpecialChars) {
    clonedRules.push({ validator: numWithSpecialCharValidator });
  }
  if (numberWithHyphen) {
    clonedRules.push({ validator: numberWithHyphenValidator });
  }
  if (charOnly) {
    clonedRules.push({ validator: stringValidator });
  }
  if (numberWithDecimal) {
    clonedRules.push({ validator: numberWithDecimalPlacesValidator });
  }
  if (naturalNumberOnly) {
    clonedRules.push({ validator: naturalNumberValidator });
  }
  if (type === "password" && applyValidation) {
    clonedRules.push({ validator: passwordValidator });
  }
  if (required) {
    clonedRules.push({ validator: requiredValidator(label) });
  }
};

const getNumberString = (numberOnly) => (numberOnly ? "digits" : "characters");

const getTextInput = (Component) => {
  const TextInputComponent = (props) => {
    const {
      label,
      value,
      name,
      required,
      labelSpan,
      inputSpan,
      rules,
      disabled,
      id,
      charOnly,
      maxValueLength,
      minValueLength,
      alphaNumericOnly,
      numberOnly,
      numbersAndSpecialChars,
      decimalPlaces,
      onClick,
      className,
      placeholder,
      autoSize,
      labelAfter,
      type,
      inputProps,
      onChange,
      onBlur,
      isFormItem,
      applyValidation,
      numberOnlyWithSign,
      updatePassword,
      numberWithHyphen,
      numberWithDecimal,
      dataTestId,
      showInputTooltip,
      isValueAfterDecimal,
      ...otherProps
    } = props;
    const clonedRules = cloneDeep(rules);
    applyAddOnRules(clonedRules, props);

    if (maxValueLength) {
      clonedRules.push({
        max: maxValueLength,
        message: `${label} cannot be longer than ${maxValueLength} ${getNumberString(
          numberOnly
        )}`,
      });
    }

    if (minValueLength) {
      clonedRules.push({
        min: minValueLength,
        message: `${label} must be at least ${minValueLength} ${getNumberString(
          numberOnly
        )}`,
      });
    }

    const setTitle = useCallback((e) => {
      const inputs = e?.currentTarget?.getElementsByTagName("input");
      if (inputs?.length) {
        Array.from(inputs).forEach((item) => {
          item.setAttribute("title", item.value);
        });
      }
    }, []);

    return (
      <div className={classNames("custom-input", className)}>
        <Row>
          <Col
            span={labelSpan}
            order={labelAfter && 1}
            className={classNames({ space: labelAfter })}
          >
            <div className="ant-form-item-label ant-form-item-label-left">
              <label
                className={classNames("ant-form-item-no-colon")}
                title={label}
                htmlFor={id}
              >
                {label}
              </label>
              <span
                className={classNames(
                  "req-star",
                  required && "ant-form-item-required"
                )}
              />
            </div>
          </Col>
          <Col
            span={inputSpan}
            onMouseEnter={showInputTooltip ? setTitle : () => {}}
          >
            {isFormItem ? (
              <Form.Item name={name} rules={[...clonedRules]} {...otherProps}>
                <Component
                  value={value}
                  name={name}
                  placeholder={placeholder}
                  autoSize={autoSize}
                  onClick={onClick}
                  maxLength={maxValueLength}
                  disabled={disabled}
                  // onKeyDown={numberOnly && handleKeyDown}
                  type={type}
                  onBlur={onBlur}
                  {...inputProps}
                  onChange={onChange}
                  data-testid={dataTestId || name}
                />
              </Form.Item>
            ) : (
              <Component
                value={value}
                name={name}
                placeholder={placeholder}
                autoSize={autoSize}
                onClick={onClick}
                maxLength={maxValueLength}
                disabled={disabled}
                // onKeyDown={numberOnly && handleKeyDown}
                type={type}
                onBlur={onBlur}
                {...otherProps}
                {...inputProps}
                onChange={onChange}
                data-testid={dataTestId || name}
              />
            )}
          </Col>
        </Row>
      </div>
    );
  };

  TextInputComponent.defaultProps = {
    label: "",
    name: "",
    rules: [],
    required: false,
    disabled: false,
    labelSpan: 10,
    inputSpan: 14,
    onClick: (event) => event.stopPropagation(),
    className: "",
    id: "",
    placeholder: "",
    value: undefined,
    charOnly: false,
    maxValueLength: null,
    minValueLength: null,
    alphaNumericOnly: false,
    numberOnly: false,
    decimalPlaces: 0,
    autoSize: {},
    labelAfter: false,
    type: "text",
    inputProps: {},
    isFormItem: true,
    onChange: () => {
      /* This is intentional */
    },
    applyValidation: true,
    dataTestId: "",
    showInputTooltip: true,
  };

  TextInputComponent.propTypes = {
    labelSpan: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    inputSpan: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    onClick: PropTypes.func,
    name: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
    className: PropTypes.string,
    label: PropTypes.string,
    required: PropTypes.bool,
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    disabled: PropTypes.bool,
    rules: PropTypes.instanceOf(Array),
    placeholder: PropTypes.string,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    charOnly: PropTypes.bool,
    maxValueLength: PropTypes.number,
    minValueLength: PropTypes.number,
    alphaNumericOnly: PropTypes.bool,
    numberOnly: PropTypes.bool,
    decimalPlaces: PropTypes.number,
    autoSize: PropTypes.instanceOf(Object),
    labelAfter: PropTypes.bool,
    type: PropTypes.string,
    inputProps: PropTypes.instanceOf(Object),
    isFormItem: PropTypes.bool,
    onChange: PropTypes.func,
    applyValidation: PropTypes.bool,
    dataTestId: PropTypes.string,
    showInputTooltip: PropTypes.bool,
  };

  return TextInputComponent;
};

const TextInput = React.memo(getTextInput(Input));
TextInput.TextArea = React.memo(getTextInput(Input.TextArea));

export default TextInput;
