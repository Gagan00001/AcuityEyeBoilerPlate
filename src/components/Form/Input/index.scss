@import "../../../css/vars";
@import "../../../css/root";
@import "../../../css/mixin";

.non-editable {
  .ant-row {
    .ant-form-item {
      margin: 0;
      .ant-form-item-control-input {
        .ant-form-item-control-input-content {
          .ant-input-disabled {
            border: none;
            cursor: pointer;
            background: none;
            padding: 0;
            font-size: $font-size-m;
            color: $color-blue-light;
            &::-webkit-input-placeholder {
              /* Chrome/Opera/Safari */
              color: transparent !important;
            }
            &::-moz-placeholder {
              /* Firefox 19+ */
              color: transparent !important;
            }
            &:-ms-input-placeholder {
              /* IE 10+ */
              color: transparent !important;
            }
            &:-moz-placeholder {
              /* Firefox 18- */
              color: transparent !important;
            }
          }
          .ant-input-number-disabled .ant-input-number-input-wrap .ant-input-number-input {
            &::-webkit-input-placeholder {
              /* Chrome/Opera/Safari */
              color: transparent !important;
            }
            &::-moz-placeholder {
              /* Firefox 19+ */
              color: transparent !important;
            }
            &:-ms-input-placeholder {
              /* IE 10+ */
              color: transparent !important;
            }
            &:-moz-placeholder {
              /* Firefox 18- */
              color: transparent !important;
            }
          }
        }
      }
    }
  }
}
.custom-input {
  .ant-row {
    .ant-form-item-label > label {
      font-size: $font-size-xsm;
      color: $color-blue-light;
    }
    .ant-form-item-control {
      .ant-input {
        border: $gap-01 solid $border-color;
        height: 2.8rem;
        border-radius: $gap-04;
        color: $color-blue-light;
        font-size: $font-size-m;
        padding-left: 6px;
        padding-right: 6px;
      }
      .ant-input:focus {
        border-color: #57a8e9;
        outline: 0;
        -webkit-box-shadow: 0 0 0 2px rgba(87, 168, 233, 0.2);
        box-shadow: 0 0 0 2px rgba(87, 168, 233, 0.2);
      }
    }
  }
  .ant-form-item-has-error {
    .ant-form-item-control-input-content {
      .ant-input {
        border-color: $color-border-red;
      }
      .ant-picker {
        border-color: $color-border-red;
      }
    }
  }
  &.non-editable {
    .ant-row {
      .ant-form-item-control {
        .ant-input,
        .ant-input-number-input {
          color: $color-blue-light;
          font-size: $font-size-m;
        }
      }
      .ant-select-single {
        .ant-select-selector {
          .ant-select-selection-item {
            font-size: $font-size-xsm;
          }
        }
      }
    }
  }
}

.ant-col.ant-col-20.ant-form-item-control {
  max-width: 100%;
}

.error-type-list {
  padding-left: 8px;
  min-width: 100px;
  li {
    // list-style: disc;
    margin-bottom: 4px;
    font-size: 1.2rem;
    position: relative;
    &::before{
      content: '';
      position: absolute;
      width: 4px;
      height: 4px;
      border-radius: 50%;
      left: -8px;
      top: 4px;
      background: #f5222d;
    }
  }
}

@media (max-width: $tablet-medium) {
  .custom-input {
    .ant-row {
      .ant-form-item-control {
        .ant-input {
          font-size: $font-size-xsm;
        }
      }
    }
  }
}
