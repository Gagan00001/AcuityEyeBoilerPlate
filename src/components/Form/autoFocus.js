function getFieldId(namePath, formName) {
  if (!namePath.length) return undefined;
  const mergedId = namePath.join('_');
  return formName ? ''.concat(formName, '_').concat(mergedId) : mergedId;
}

function toArray(candidate) {
  if (candidate === undefined || candidate === false) return [];
  return Array.isArray(candidate) ? candidate : [candidate];
}

export default function autoFocus(name, form) {
  const namePath = toArray(name);
  // eslint-disable-next-line no-underscore-dangle
  const fieldId = getFieldId(namePath, form && form.__INTERNAL__ && form.__INTERNAL__.name);
  const node = fieldId ? document.getElementById(fieldId) : null;
  if (node) {
    node.focus();
  }
}
