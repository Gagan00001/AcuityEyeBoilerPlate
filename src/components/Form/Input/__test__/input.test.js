/* eslint-disable no-undef */
import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import { Form as AntdForm } from 'antd';
import userEvent from '@testing-library/user-event';
import Input from '../index';
import Form from '../../index';
import WithAppWrapper from '../../../../hoc/withAppWrapper';

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const RenderInputComponent = (props) => {
  const [form] = AntdForm.useForm();
  return (
    <Form form={form}>
      <Input
        {...props}
      />
    </Form>
  );
};

const otherProps = {
  rules: [],
  labelSpan: 8,
  inputSpan: 16,
  disabled: false,
  maxValueLength: 20,
  minValueLength: 1,
  onClick: jest.fn(),
  className: '',
  placeholder: 'Test Input',
  autoSize: {},
  inputProps: {},
  onChange: jest.fn(),
  onBlur: jest.fn(),
  alphaNumericOnly: true,
  numberOnlyWithSign: true,
  numberOnlySignValidator: jest.fn(),
  alphaNumericValidator: jest.fn(),
  decimalPlaces: 1,
  updatePassword: true,
  numberWithHyphen: true,
  charOnly: true,
  type: 'password',
  applyValidation: true,
};

describe('should match input form item snapshot', () => {
  const WrappedRadioComponent = WithAppWrapper(RenderInputComponent);
  test('should match snapshot', () => {
    const { container: inputContainer } = render(<WrappedRadioComponent
      {...otherProps}
      label="Test Input"
      name="testInput"
      id="test-input"
      dataTestId="test-input"
    />, container);
    expect(inputContainer).toMatchSnapshot();
  });

  test('should match input snapshot without form item', () => {
    const { container: inputContainer } = render(<WrappedRadioComponent
      {...otherProps}
      label="Test Input"
      name="testInput"
      id="test-input"
      dataTestId="test-input"
      isFormItem={false}
    />, container);
    expect(inputContainer).toMatchSnapshot();
  });

  test('should render input', () => {
    const {
      getByTestId,
    } = render(<WrappedRadioComponent
      label="Test Input"
      name="testInput"
      id="test-input"
      dataTestId="test-input"
    />, container);
    expect(getByTestId('test-input')).toBeInTheDocument();
  });

  test('should be required when passed required', () => {
    render(<WrappedRadioComponent
      label="Test Input"
      name="testInput"
      required
      id="test-input"
      dataTestId="test-input"
    />, container);
    expect(document.querySelector('.ant-form-item-required')).toBeInTheDocument();
  });

  test('should be able to change value', () => {
    const {
      getByTestId,
    } = render(<WrappedRadioComponent
      label="Test Input"
      name="testInput"
      required
      id="test-input"
      dataTestId="test-input"
    />, container);
    const input = getByTestId('test-input');
    userEvent.click(input);
    fireEvent.change(input, { target: { value: 'testing input value' } });
    expect(input.value).toEqual('testing input value');
  });

  test('should be able to clear value', () => {
    const {
      getByTestId,
    } = render(<WrappedRadioComponent
      label="Test Input"
      name="testInput"
      required
      id="test-input"
      dataTestId="test-input"
    />, container);
    const input = getByTestId('test-input');
    userEvent.click(input);
    fireEvent.change(input, { target: { value: 'testing input value' } });
    expect(input.value).toEqual('testing input value');
    userEvent.click(input);
    userEvent.clear(input);
    expect(input.value).toEqual('');
  });

  test('should be able to type number only', () => {
    const {
      getByTestId,
    } = render(<WrappedRadioComponent
      label="Test Input"
      name="testInput"
      required
      id="test-input"
      dataTestId="test-input"
      alphaNumericOnly={false}
      numberOnlyWithSign={false}
      numberOnly
      decimalPlaces={1}
    />, container);
    const input = getByTestId('test-input');
    userEvent.click(input);
    fireEvent.blur(input, { target: { value: undefined } });
    userEvent.type(input, 'testing input value 1234');
    userEvent.click(input);
    expect(input.value?.trim()).toEqual('1234');
  });

  test('should be able to type special characters', () => {
    const {
      getByTestId,
    } = render(<WrappedRadioComponent
      label="Test Input"
      name="testInput"
      required
      id="test-input"
      dataTestId="test-input"
      alphaNumericOnly={false}
      numberOnlyWithSign={false}
      numbersAndSpecialChars
    />, container);
    const input = getByTestId('test-input');
    userEvent.click(input);
    userEvent.type(input, '1234#');
    expect(input.value?.trim()).toEqual('1234#');
  });
});
