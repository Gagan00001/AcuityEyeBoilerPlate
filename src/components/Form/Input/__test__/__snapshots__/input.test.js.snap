// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match input form item snapshot should match input snapshot without form item 1`] = `
<div>
  <div
    style="flex: 1;"
  >
    <div>
      <form
        class="ant-form ant-form-horizontal ant-form-hide-required-mark ant-form-middle"
        id="form"
      >
        <div
          class="sc-AxjAm hGbGhh form-grid-component"
        >
          <div
            class="custom-input"
          >
            <div
              class="ant-row"
            >
              <div
                class="ant-col ant-col-8"
              >
                <div
                  class="ant-form-item-label ant-form-item-label-left"
                >
                  <label
                    class="ant-form-item-no-colon"
                    for="test-input"
                    title="Test Input"
                  >
                    Test Input
                  </label>
                  <span
                    class="req-star"
                  />
                </div>
              </div>
              <div
                class="ant-col ant-col-16"
              >
                <input
                  autosize="[object Object]"
                  class="ant-input"
                  data-testid="test-input"
                  maxlength="20"
                  name="testInput"
                  placeholder="Test Input"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`should match input form item snapshot should match snapshot 1`] = `
<div>
  <div
    style="flex: 1;"
  >
    <div>
      <form
        class="ant-form ant-form-horizontal ant-form-hide-required-mark ant-form-middle"
        id="form"
      >
        <div
          class="sc-AxjAm hGbGhh form-grid-component"
        >
          <div
            class="custom-input"
          >
            <div
              class="ant-row"
            >
              <div
                class="ant-col ant-col-8"
              >
                <div
                  class="ant-form-item-label ant-form-item-label-left"
                >
                  <label
                    class="ant-form-item-no-colon"
                    for="test-input"
                    title="Test Input"
                  >
                    Test Input
                  </label>
                  <span
                    class="req-star"
                  />
                </div>
              </div>
              <div
                class="ant-col ant-col-16"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-col-20 ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          autosize="[object Object]"
                          class="ant-input"
                          data-testid="test-input"
                          id="form_testInput"
                          maxlength="20"
                          name="testInput"
                          placeholder="Test Input"
                          type="password"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
